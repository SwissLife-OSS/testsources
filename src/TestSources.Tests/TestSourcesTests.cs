using System.IO;
using Snapshooter.Xunit;
using Xunit;
using FluentAssertions;

namespace TestSources.Tests
{
    public class TestSourcesTests
    {
        [Fact]
        public void ExistingFile_CanBeFound()
        {
            // arrange
            TestSources ts = new TestSources();

            //act



            //assert
            FileInfo fi = new FileInfo("C:\\GIT\testsources\\src\\TestSources.Tests\\__TestSources__\\sub01\\SubTextFile01.txt");
            DirectoryInfo di = new DirectoryInfo("C:\\GIT\\testsources\\src\\TestSources.Tests\\__TestSources__");
            //FileSystemInfo fsi = new FileSystemInfo();

            // Name , the dir or file name
            var xxname = fi.Name; // or di.Name;
            //File f = new File()

            // Instrad of DirectoryName, we have FullName
            var xxx = fi.FullName; // or di.FullName;
            var xxxx = di.FullName;

            Assert.NotNull(fi);
            Assert.NotNull(di);
        }


        [Fact]
        public void TestSources_Constructor_works()
        {
            // Arrange
            TestSources testSources = new TestSources();

            // Act
            // Assert
            testSources.FilesAndFolders.MatchSnapshot();
            testSources.Name.Should().Be("__Testsources__");
            testSources.Parent.Should().BeNull();
        }
    }
}
