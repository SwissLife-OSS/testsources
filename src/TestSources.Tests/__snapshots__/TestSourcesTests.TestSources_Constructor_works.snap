[
  {
    "Name": "BinFile01.rar",
    "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\BinFile01.rar",
    "Parent": [
      {
        "Name": "BinFile02.rar",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\BinFile02.rar"
      },
      {
        "Name": "TextFile01.txt",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\TextFile01.txt"
      },
      {
        "Name": "TextFile02.txt",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\TextFile02.txt"
      },
      [
        {
          "Name": "SomeJson.json",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\FileContentTests\\SomeJson.json"
        },
        {
          "Name": "SomeJson.rar",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\FileContentTests\\SomeJson.rar"
        }
      ],
      [
        {
          "Name": "SubBinFile01.rar",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubBinFile01.rar"
        },
        {
          "Name": "SubTextFile01.txt",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubTextFile01.txt"
        },
        {
          "Name": "SubTextFile02.txt",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubTextFile02.txt"
        },
        [
          {
            "Name": "Sub2FolderBinFile.rar",
            "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2FolderBinFile.rar"
          },
          {
            "Name": "Sub2TextFile.txt",
            "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2TextFile.txt"
          }
        ]
      ]
    ]
  },
  {
    "Name": "BinFile02.rar",
    "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\BinFile02.rar",
    "Parent": [
      {
        "Name": "BinFile01.rar",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\BinFile01.rar"
      },
      {
        "Name": "TextFile01.txt",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\TextFile01.txt"
      },
      {
        "Name": "TextFile02.txt",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\TextFile02.txt"
      },
      [
        {
          "Name": "SomeJson.json",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\FileContentTests\\SomeJson.json"
        },
        {
          "Name": "SomeJson.rar",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\FileContentTests\\SomeJson.rar"
        }
      ],
      [
        {
          "Name": "SubBinFile01.rar",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubBinFile01.rar"
        },
        {
          "Name": "SubTextFile01.txt",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubTextFile01.txt"
        },
        {
          "Name": "SubTextFile02.txt",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubTextFile02.txt"
        },
        [
          {
            "Name": "Sub2FolderBinFile.rar",
            "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2FolderBinFile.rar"
          },
          {
            "Name": "Sub2TextFile.txt",
            "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2TextFile.txt"
          }
        ]
      ]
    ]
  },
  {
    "Name": "TextFile01.txt",
    "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\TextFile01.txt",
    "Parent": [
      {
        "Name": "BinFile01.rar",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\BinFile01.rar"
      },
      {
        "Name": "BinFile02.rar",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\BinFile02.rar"
      },
      {
        "Name": "TextFile02.txt",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\TextFile02.txt"
      },
      [
        {
          "Name": "SomeJson.json",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\FileContentTests\\SomeJson.json"
        },
        {
          "Name": "SomeJson.rar",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\FileContentTests\\SomeJson.rar"
        }
      ],
      [
        {
          "Name": "SubBinFile01.rar",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubBinFile01.rar"
        },
        {
          "Name": "SubTextFile01.txt",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubTextFile01.txt"
        },
        {
          "Name": "SubTextFile02.txt",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubTextFile02.txt"
        },
        [
          {
            "Name": "Sub2FolderBinFile.rar",
            "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2FolderBinFile.rar"
          },
          {
            "Name": "Sub2TextFile.txt",
            "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2TextFile.txt"
          }
        ]
      ]
    ]
  },
  {
    "Name": "TextFile02.txt",
    "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\TextFile02.txt",
    "Parent": [
      {
        "Name": "BinFile01.rar",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\BinFile01.rar"
      },
      {
        "Name": "BinFile02.rar",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\BinFile02.rar"
      },
      {
        "Name": "TextFile01.txt",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\TextFile01.txt"
      },
      [
        {
          "Name": "SomeJson.json",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\FileContentTests\\SomeJson.json"
        },
        {
          "Name": "SomeJson.rar",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\FileContentTests\\SomeJson.rar"
        }
      ],
      [
        {
          "Name": "SubBinFile01.rar",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubBinFile01.rar"
        },
        {
          "Name": "SubTextFile01.txt",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubTextFile01.txt"
        },
        {
          "Name": "SubTextFile02.txt",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubTextFile02.txt"
        },
        [
          {
            "Name": "Sub2FolderBinFile.rar",
            "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2FolderBinFile.rar"
          },
          {
            "Name": "Sub2TextFile.txt",
            "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2TextFile.txt"
          }
        ]
      ]
    ]
  },
  [
    {
      "Name": "SomeJson.json",
      "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\FileContentTests\\SomeJson.json"
    },
    {
      "Name": "SomeJson.rar",
      "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\FileContentTests\\SomeJson.rar"
    }
  ],
  {
    "Name": "SomeJson.json",
    "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\FileContentTests\\SomeJson.json",
    "Parent": [
      {
        "Name": "SomeJson.rar",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\FileContentTests\\SomeJson.rar"
      }
    ]
  },
  {
    "Name": "SomeJson.rar",
    "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\FileContentTests\\SomeJson.rar",
    "Parent": [
      {
        "Name": "SomeJson.json",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\FileContentTests\\SomeJson.json"
      }
    ]
  },
  [
    {
      "Name": "SubBinFile01.rar",
      "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubBinFile01.rar"
    },
    {
      "Name": "SubTextFile01.txt",
      "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubTextFile01.txt"
    },
    {
      "Name": "SubTextFile02.txt",
      "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubTextFile02.txt"
    },
    [
      {
        "Name": "Sub2FolderBinFile.rar",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2FolderBinFile.rar"
      },
      {
        "Name": "Sub2TextFile.txt",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2TextFile.txt"
      }
    ]
  ],
  {
    "Name": "SubBinFile01.rar",
    "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubBinFile01.rar",
    "Parent": [
      {
        "Name": "SubTextFile01.txt",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubTextFile01.txt"
      },
      {
        "Name": "SubTextFile02.txt",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubTextFile02.txt"
      },
      [
        {
          "Name": "Sub2FolderBinFile.rar",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2FolderBinFile.rar"
        },
        {
          "Name": "Sub2TextFile.txt",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2TextFile.txt"
        }
      ]
    ]
  },
  {
    "Name": "SubTextFile01.txt",
    "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubTextFile01.txt",
    "Parent": [
      {
        "Name": "SubBinFile01.rar",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubBinFile01.rar"
      },
      {
        "Name": "SubTextFile02.txt",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubTextFile02.txt"
      },
      [
        {
          "Name": "Sub2FolderBinFile.rar",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2FolderBinFile.rar"
        },
        {
          "Name": "Sub2TextFile.txt",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2TextFile.txt"
        }
      ]
    ]
  },
  {
    "Name": "SubTextFile02.txt",
    "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubTextFile02.txt",
    "Parent": [
      {
        "Name": "SubBinFile01.rar",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubBinFile01.rar"
      },
      {
        "Name": "SubTextFile01.txt",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\SubTextFile01.txt"
      },
      [
        {
          "Name": "Sub2FolderBinFile.rar",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2FolderBinFile.rar"
        },
        {
          "Name": "Sub2TextFile.txt",
          "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2TextFile.txt"
        }
      ]
    ]
  },
  [
    {
      "Name": "Sub2FolderBinFile.rar",
      "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2FolderBinFile.rar"
    },
    {
      "Name": "Sub2TextFile.txt",
      "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2TextFile.txt"
    }
  ],
  {
    "Name": "Sub2FolderBinFile.rar",
    "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2FolderBinFile.rar",
    "Parent": [
      {
        "Name": "Sub2TextFile.txt",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2TextFile.txt"
      }
    ]
  },
  {
    "Name": "Sub2TextFile.txt",
    "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2TextFile.txt",
    "Parent": [
      {
        "Name": "Sub2FolderBinFile.rar",
        "FullName": "C:\\git\\testsources\\src\\TestSources.Tests\\__testsources__\\sub01\\sub02\\Sub2FolderBinFile.rar"
      }
    ]
  }
]
